common/flatpak-context.c

```diff
@@ -915,41 +953,17 @@ flatpak_context_merge (FlatpakContext *context,
   while (g_hash_table_iter_next (&iter, &key, &value))
     g_hash_table_insert (context->persistent, g_strdup (key), value);
 
-  /* We first handle all negative home and host as they override other
-     keys than themselves from the parent */
-  if (g_hash_table_lookup_extended (other->filesystems,
-                                    "host",
-                                    NULL, &value))
-    {
-      FlatpakFilesystemMode host_mode = GPOINTER_TO_INT (value);
-      if (host_mode == FLATPAK_FILESYSTEM_MODE_NONE)
-        no_host = TRUE;
-    }
+  if (g_hash_table_lookup_extended (other->filesystems, "host-reset", NULL, &value))
+    fs_host_reset = GPOINTER_TO_INT (value) == FLATPAK_FILESYSTEM_MODE_NONE;
```

Comment:
This seems odd: what happens if it's stored with a different value?

The least-bad semantics would probably be for `FLATPAK_FILESYSTEM_MODE_READ_ONLY` to limit inherited permissions to being read-only, and for `FLATPAK_FILESYSTEM_MODE_READ_WRITE` to apply no limits - but because there's no correct way to get those, I would be inclined to throw in a `g_warn_if_fail()` instead, or something.

Reply:
It now does a `g_warn_if_fail()`.